profiler/src/ProfilerEngine/Datadog.Profiler.Native/ActivityHelpers.cpp

```diff
@@ -0,0 +1,162 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "ActivityHelpers.h"
+
+bool ActivityHelpers::IsActivityPath(const GUID* pActivityId, int processID)
+{
+    uint32_t* uintPtr = (uint32_t*)pActivityId;
+
+    uint32_t sum = uintPtr[0] + uintPtr[1] + uintPtr[2] + 0x599D99AD;
+    if (processID == 0)
+    {
+        // We guess that the process ID is < 20 bits and because it was xored
+        // with the lower bits, the upper 12 bits should be independent of the
+        // particular process, so we can at least confirm that the upper bits
+        // match.
+        return ((sum & 0xFFF00000) == (uintPtr[3] & 0xFFF00000));
+    }
+
+    if ((sum ^ (uint32_t)processID) == uintPtr[3])  // This is the new style
+    {
+        return true;
+    }
+
+    return (sum == uintPtr[3]);         // THis is old style where we don't make the ID unique machine wide.
+}
+
+int ActivityHelpers::ActivityPathProcessID(const GUID* pActivityId)
```

Comment:
GetProcessIdFromActivatyPath ?

Reply:
Let's use the `GetProcess` shorter name since the `ActivityPath` is implict by the parameter